packages/orchestration/src/exos/stakingAccountKit.js

```diff
@@ -149,6 +240,49 @@ export const prepareStakingAccountKit = (baggage, makeRecorderKit, zcf) => {
             return this.facets.holder.delegate(validator, amount);
           }, 'Delegate');
         },
+        Deposit() {
+          trace('Deposit');
+          // TODO consider adding timeoutTimestamp as a parameter. current
+          // default is `FIVE_MINUTES_IN_SECONDS`
+          return zcf.makeInvitation(
+            async seat => {
+              const { give } = seat.getProposal();
+              // only one entry permitted by proposal shape
+              const [keyword, giveAmount] = Object.entries(give)[0];
+              this.state.brandToIssuer.has(giveAmount.brand) ||
+                Fail`${giveAmount.brand} not registered`;
+
+              const payments = await withdrawFromSeat(zcf, seat, give);
+              const payment = await Object.values(payments)[0];
+              try {
+                await this.facets.holder.deposit(payment);
+              } catch (_depositOrTransferError) {
+                try {
+                  await depositToSeat(zcf, seat, give, payments);
+                  throw Fail`Deposit failed, payment returned.`;
+                } catch (error) {
+                  if (error.message.includes('not a live paymen')) {
```

Comment:
```suggestion
                  if (error.message.includes('not a live payment')) {
```

but this is brittle. is there no more reliable signal?

Reply:
Our [style for errors and control flow](https://github.com/Agoric/agoric-sdk/wiki/Errors-and-Control-Flow) is to not peek at errors.

Why do we care what sort of error it is?

Reply:
Also, are we testing all these failure modes?

I wonder if it would help to hoist much of this logic to a function that we can unit test.